pkg/proxy/logger/endpoint/epinfo.go

```diff
@@ -40,42 +40,45 @@ func NewEndpointInfoRegistry(ipc *ipcache.IPCache, endpointManager endpointmanag
 	}
 }
 
-func (r *endpointInfoRegistry) FillEndpointInfo(info *accesslog.EndpointInfo, addr netip.Addr, id identity.NumericIdentity) {
+func (r *endpointInfoRegistry) FillEndpointInfo(info *accesslog.EndpointInfo, addr netip.Addr) {
 	var ep *endpoint.Endpoint
 	if addr.IsValid() {
 		if addr.Is4() {
 			info.IPv4 = addr.String()
 		} else {
 			info.IPv6 = addr.String()
 		}
-
-		// Get (local) endpoint identifier to be reported by cilium monitor
-		ep = r.endpointManager.LookupIP(addr)
-		if ep != nil {
-			info.ID = ep.GetID()
-		}
 	}
 
 	// Only resolve the security identity if not passed in, as it may have changed since
 	// reported by the proxy. This way we log the security identity and labels used for
 	// policy enforcement, if any.
-	if id == 0 {
+	if info.ID == 0 || info.Identity == 0 {
```

Comment:
If `ID` is equal to zero, is `Identity` also always equal to 0? In the case that only one is not set, we'd waste time updating both.

Reply:
Good question. The unfortunately-named ID is actually the Endpoint ID, not the security Identity. There are cases in the proxy where we already know the remote identity, but not the endpoint.

Of course, we can't know the endpoint ID if it is remote, but if we proxy a DNS request to coredns on the same node, then it is possible to add the endpoint ID as well.

Reply:
However, you are right, the conditions aren't clear. I made the logic nicer.